### ISSUE TITLE
[Question] tests are skipped randomly

### ISSUE THREAD

**Issue by subhra-s-das** [microsoft/playwright#20350]
Out of 2500 tests which runs nightly using Jenkins some tests are skipped randomly some nights. Tests are designed to run in parallel using 4 workers at any given time `npx cross-env  PW_PROJECT=default playwright test --project default --workers 4`. Some test files has just one `describe` block but few has multiple `describe` , and each `describe` has it's own `before` and `after` hooks. Test having a single `describe` deals with one Web form and those having multiple `describe` deals with multiple Web forms. 

**What is the problem :** 
Line 16 shows test `C3885287` starts on `worker1 # 2` which failed.

Line 38 shows a new worker `worker1 #4` created for the next text `C3885288`

and from line 51 remaining tests `C3885289, C3885290, C3885291, C3885292, C3885305 ` from same  `describe` block 
got skipped as no new `worker ID` is assigned

```
1.	Running 40 tests using 4 workers
2.	
3.	Running 40 tests using 4 workers
4.	
5.	globalSetup login with username web
6.	     1 [default] › credentials\credentials.e2e.ts:30:6 › TestApp credentials form › C3063856 should load credentials page and display title
7.	     2 [default] › form-module\date-picker\date-picker.e2e.ts:24:9 › TestApp date-picker module › C3371764 should create member Inv Profile records
8.	     3 [default] › member-investment-profile\member-investment-profile.e2e.ts:34:6 › TestApp member investment profile form › C3224831 should load member investment profile form and display title
9.	     4 [default] › form-module\filters\filters.e2e.ts:38:6 › TestApp filters module - member balance history › C3885287 should give correct filter result for filter operator starts with lower case
10.	worker #1, title: C3371764 should create member Inv Profile records
11.	worker #1, title: C3371764 should create member Inv Profile records
12.	  ok 1 [default] › credentials\credentials.e2e.ts:30:6 › TestApp credentials form › C3063856 should load credentials page and display title (5s)
13.	worker #3, title: C3224831 should load member investment profile form and display title
14.	worker #0, title: C3063856 should load credentials page and display title
15.	worker #0, title: C3063856 should load credentials page and display title
16.	worker1 #2, title: C3885287 should give correct filter result for filter operator starts with lower case
17.	worker1 #2, title: C3885287 should give correct filter result for filter operator starts with lower case
18.	  x  4 [default] › form-module\filters\filters.e2e.ts:38:6 › TestApp filters module - member balance history › C3885287 should give correct filter result for filter operator starts with lower case (11s)
19.	  x  2 [default] › form-module\date-picker\date-picker.e2e.ts:24:9 › TestApp date-picker module › C3371764 should create member Inv Profile records (15s)
20.	worker #3, C3680525 audit: should be able to access detailed audit for action type insert
21.	worker #3, C3680525 audit: should be able to access detailed audit for action type insert
22.	worker #3, C3680525 audit: should be able to access detailed audit for action type insert
23.	worker #3, C3680525 audit: should be able to access detailed audit for action type insert
24.	  ok 7 [default] › credentials\credentials.e2e.ts:34:6 › TestApp credentials form › C3063857 should display headings (3s)
25.	     8 [default] › credentials\credentials.e2e.ts:38:6 › TestApp credentials form › C3063858 should have add new credentials button
26.	ERROR No search results for "0000055" ...
27.	  ok 6 [default] › member-salary-details\member-salary-details.e2e.ts:58:6 › TestApp member salary details form › C3680525 audit: should be able to access detailed audit for action type insert (23s)
28.	search error for :0000055 exception: TimeoutError
29.	search error for :0000055 exception: TimeoutError
30.	     9 [default] › member-tax-flags-history\member-tax-flags-history.e2e.ts:31:6 › TestApp member tax flag history form › C3295063 should load member tax file history form and display title
31.	  ok 9 [default] › member-tax-flags-history\member-tax-flags-history.e2e.ts:31:6 › TestApp member tax flag history form › C3295063 should load member tax file history form and display title (2s)
32.	worker #0, C3295063 should load member tax file history form and display title
33.	  ok 10 [default] › member-tax-flags-history\member-tax-flags-history.e2e.ts:37:6 › TestApp member tax flag history form › C3295064 should display heading (2s)
34.	     11 [default] › form-module\filters\filters.e2e.ts:47:6 › TestApp filters module - member balance history › C3885288 should give correct filter result for filter operator starts with invalid character
35.	     12 [default] › personal-details\personal-details.e2e.ts:25:6 › TestApp personal details form › C4197316 should load personal details page and display title
36.	worker #0, C4197316 should load personal details page and display title
37.	  ok 12 [default] › personal-details\personal-details.e2e.ts:25:6 › TestApp personal details form › C4197316 should load personal details page and display title (2s)
38.	worker1 #4, title: C3885288 should give correct filter result for filter operator starts with invalid character
39.	  x  11 [default] › form-module\filters\filters.e2e.ts:47:6 › TestApp filters module - member balance history › C3885288 should give correct filter result for filter operator starts with invalid character (605ms)
40.	  ok 13 [default] › personal-details\personal-details.e2e.ts:31:6 › TestApp personal details form › C4197317 should display heading (2s)
41.	     14 [default] › user-details\user-details.e2e.ts:30:6 › TestApp user details form › C4213460 should load user details form and display title
42.	  ok 14 [default] › user-details\user-details.e2e.ts:30:6 › TestApp user details form › C4213460 should load user details form and display title (2s)
43.	worker #0, title: C4213460 should load user details form and display title
44.	  ok 15 [default] › user-details\user-details.e2e.ts:34:6 › TestApp user details form › C4213461 should display heading (3s)
45.	     16 [default] › form-module\date-picker\date-picker.e2e.ts:31:9 › TestApp date-picker module › C3686787 should display tooltip for date picker input field
46.	     17 [default] › member-salary-details\member-salary-details.e2e.ts:106:6 › TestApp member salary details form › C3680526 audit: should be able able to access detailed audit for action type delete
47.	worker #5, title: C3686787 should display tooltip for date picker input field
48.	worker #5, title: C3686787 should display tooltip for date picker input field
49.	  ok 17 [default] › member-salary-details\member-salary-details.e2e.ts:106:6 › TestApp member salary details form › C3680526 audit: should be able able to access detailed audit for action type delete (11s)
50.	  ok 16 [default] › form-module\date-picker\date-picker.e2e.ts:31:9 › TestApp date-picker module › C3686787 should display tooltip for date picker input field (2s)
51.	  -  18 [default] › form-module\filters\filters.e2e.ts:54:6 › TestApp filters module - member balance history › C3885289 should give correct filter result for filter operator starts with invalid number
52.	  -  19 [default] › form-module\filters\filters.e2e.ts:61:6 › TestApp filters module - member balance history › C3885290 should give correct filter result for filter operator equals ignore case
53.	  -  20 [default] › form-module\filters\filters.e2e.ts:69:6 › TestApp filters module - member balance history › C3885291 should give correct filter result for filter operator equals and is less than
54.	  -  21 [default] › form-module\filters\filters.e2e.ts:77:6 › TestApp filters module - member balance history › C3885292 should give correct filter result for operator is in
55.	  -  22 [default] › form-module\filters\filters.e2e.ts:84:6 › TestApp filters module - member balance history › C3885305 should reset filter on clear
56.	     23 [default] › form-module\filters\filters.e2e.ts:122:6 › TestApp filters module - payroll details › C3885293 should give correct filter result for operator wildcard type 1
57.	worker2 #6, title: C3885293 should give correct filter result for operator wildcard type 1
58.	worker2 #6, title: C3885293 should give correct filter result for operator wildcard type 1
59.	  x  23 [default] › form-module\filters\filters.e2e.ts:122:6 › TestApp filters module - payroll details › C3885293 should give correct filter result for operator wildcard type 1 (9s)
60.	ERROR No search results for "0000055" ...
61.	search error for :0000055 exception: TimeoutError
62.	search error for :0000055 exception: TimeoutError
```

Test Snippet 

![image](https://user-images.githubusercontent.com/14076280/214447275-7473d06b-9e0f-411d-819b-67da793cabb5.png)

**Comment by pavelfeldman** [comment]
I'm not sure how to parse the output. If you are seeing that some tests don't run, this is likely due to the globalTimeout that you have for your test run. Starting it with one of the built-in reporters will show the global timeout error message.

**Comment by pavelfeldman** [comment]
Another idea is that your beforeAll fails - if beforeAll fails, you'll see an error message and the tests that need it won't run. Overall, it is somewhat hard to diagnose it based on the screenshots that show some code, so we would need to have exact steps to follow or at the very least, a complete terminal output when a stock reporter is used.

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.

**Comment by rostislav-simonik-plc** [comment]
@pavelfeldman is it possible to override the behaviour somehow all tests will actually report as failed if beforeall fails, i understand the concept, but it's actually creating confusion in our team, and we would probably prefer failure instead.
